### DISCUSSION TITLE
Goal of the project?

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#8] [General]
Should it be an installable package or a "framework" showcasing best practices and cool stuff (like hybrid framework back in the days)

**Comment by NikRimington** [comment]
There are pro's and con's for both, if you are going for an installable package, are you looking to include doctypes and the like?
Can the project have other dependencies, such as usync for example.

Personally, I think the output should be more framework based, or even a dotnet new template, a bit like what the Packages team is creating to help package devs. A cool "starter project" as opposed to "starter kit" that people can start their projects with would be very nice. Opens up to evolving it for vNext when that comes out as well.

**Reply by CarlSargunar** [reply]
I like the idea of a template. Never built one, but a new template would be awesome if we can wrap all the functionality we want to build right into it

**Reply by CarlSargunar** [reply]
How would it work with things like dependancies on Umbraco versions? What about database and breaking changes (if there are any)? I presume there's a way of supporting a list of supported umbraco versions?

**Reply by TimGeyssens** [reply]
Good Idea Nik, dunno if this is an option: fork the solution and then vs asks for the new solution name... renaming the projects and namespaces....

**Comment by Mantus667** [comment]
The framework option is a nice thing but I guess it would be used like HybridFramework, so more like a reference where you could get inspiration from.
The starterkit thing would be nice as I kind of dislike the current one. That said I doubt that starter kits are really used from anybody. So it would have to be more like a solution as uSkinned provides. So complete website bundles installable as package. And I think that is not what should be achieved here.
The starter project idea by Nik is great as I would love to get some insights on how to create templates for dotnet new. Which is more like a personal preference. And kind of an extension to the other options above.

I personally would tend to a starter kit. If the outcome is kind of a best practices approach to build some of the most common things like a news area, I would at least package it up.

Makes sense? I hope...

**Reply by TimGeyssens** [reply]
maybe start as starter project and also package bits so it outputs a starterkit?

**Comment by marcemarc** [comment]
Remember "there is no 'best practice' - only 'good practice in context' " (reference: https://twitter.com/munkimagik), so you can get caught out trying to be opinionated on 'how to do things', in different circumstances - eg(what IF they switch to an Oracle db? what if there there is a 100,000 content nodes, will this be slow etc etc, you can under optimise or over optimise - 'be simple' or 'over complicate' it is the context of the site you are delivering that makes (hopefully) the right decision on approach in the real world but with multiple contributors to a thing like this and without a 'context' to hang it on, there's going to be a whole lot of discussion.

So my suggestion would be to define a real world example, to build a 'good' Umbraco site for to show off multiple good patterns, and continue to update it over time, but more importantly to establish a context that enables you to make decisions.

The official documentation could reference tutorials and examples from it.

An example? 

The local government starter kit was always great, as it was well implemented and you immediately understood the context the site was trying to deliver and lots of different types of pages to show off different Umbraco functionality etc

But maybe it needs to be more fun?

Say you are building a website for an 'Artist', called umbanksy or something, they would start with a basic profile page... maybe they write a blog.... maybe they share images / videos in a gallery of their work... have Instagram accounts to integrate - maybe they do live events, merchandise to sell, or art to sell, maybe they appear on reality TV, and suddenly the site needs to scale etc etc - 

Your hackathon could establish the basics, and the profile page - different people could branch off to build different bits of functionality, it would mirror a real site, you wouldn't need to build it all in one hackathon, - but try to make it 'shippable' at the end of each 'push' - over time you could show it being refactored etc etc

But the main thing is by having a deadline to ship for a 'real' world thing, will enable decisions to be made in a context, which my theory is, might prevent any blockages in discussion, or incomplete work at the end...

... hey and real 'artists' might one day use it...

**Reply by TimGeyssens** [reply]
good idea! Let's take it over to the #16 real world use case, my intial idea was to do a real estate agent, since that also has some nice searching/filtering...

**Comment by dawoe** [comment]
Personally I would go the Hybrid Framework route. I always used starterkits as learning resources. I have never built a site based off a starterkit and probably never will. 
Like @NikRimington this could evolve to a dotnet template. So with different options in the command line you can add different functionality to your solution....

**Reply by TimGeyssens** [reply]
yes tend to agree, framework but use a specific use case... if there is need we can package it up to a starter kit but not a prio

**Comment by nathanwoulfe** [comment]
Try not to get caught up in what tools are used - frameworks and task runners are great, but if the aim is to build a great examplar Umbraco site, Umbraco should be front and centre. 

Different devs will favour their favourite tools, this project should be highlighting that Umbraco doesn't care what you use to build the front end. 

Focus on showcasing Umbraco.

**Reply by louisjrdev** [reply]
I see this in the opposite way too though, Umbraco doesn't need pushing to a community who is already aware of Umbraco and how to build it. Where as if Umbraco were showcased with modern frameworks and tools, then the users of those tools and frameworks could see how seamless it is to integrate with Umbraco and what it is capable of. 

No snide intended, but an MVC site with some vanilla css and js, is not exactly mind blowing.. people think what they see is all it can do most of the time
